torch/distributed/_tensor/_dispatch.py

```diff
@@ -307,15 +308,17 @@ def unwrap_to_op_info(
         local_kwargs: Dict[str, object] = {}
         mesh: Optional[DeviceMesh] = None
 
+        # record the largest mesh only if implicit replication is turned on
+        if self._allow_implicit_replication:
+            for arg in chain(iter(args_list), iter(kwargs.values())):
```

Comment:
I wonder how large is this list in general, e.g. for `foreach`. Do you think it'd be helpful if we do pruning / early break here? E.g. we can get the root mesh, and if the ndim is the same as root mesh, we know it's root mesh and break.

Reply:
> I wonder how large is this list in general, e.g. for `foreach`. Do you think it'd be helpful if we do pruning / early break here? E.g. we can get the root mesh, and if the ndim is the same as root mesh, we know it's root mesh and break.

True. This is more generic, but since the current use case is for 2D use case only, I guess we can break out of the loop when we get a mesh with ndim == 2 and add a comment here. If later on, we see we need to support more general cases, we can remove the early break.